File: python3/easy/temperatures.py
import sys
import math

n = int(input())
numbers = input().split()
if not numbers:
    print(0)
else:
    pos = min([int(i) if int(i) >= 0 else math.inf for i in numbers])
    neg = max([int(i) if int(i) < 0 else (math.inf * -1) for i in numbers])
    print(pos if pos <= abs(neg) else neg)

File: javascript/easy/temperatures.js
readline();

closest = readline()
    .split(' ')
    .map((n) => + n)
    .reduce((accumulator, current) => {
        if (current < 0) {
            accumulator[0] = Math.max(accumulator[0], current);
        } else {
            accumulator[1] = Math.min(accumulator[1], current);
        }
        return accumulator;
    }, [-Infinity, Infinity]);

print(Math.abs(closest[0]) < closest[1] ? closest[0] : closest[1]);
